addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/api/manager/EnergenieApiManagerImpl.java

```diff
@@ -0,0 +1,235 @@
+        parameters.put(DEVICE_ID_KEY, id);
+        // With this parameter we exclude the historical usage data ("1" - to include, "0" - to exclude)
+        // The historical data is from no importance to the user,
+        // it consists of the number of events for each hour from the last 24 hours.
+        parameters.put(SUBDEVICE_INCLUDE_USAGE_DATA, 0);
+        JsonObject result = execute(CONTROLLER_SUBDEVICES, ACTION_SHOW, parameters);
+        return JsonResponseHandler.getObject(result, JsonSubdevice.class);
+    }
+
+    @Override
+    public JsonSubdevice updateSubdevice(int id, String label) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DEVICE_ID_KEY, id);
+        parameters.put(DEVICE_LABEL_KEY, label);
+        JsonObject result = execute(CONTROLLER_SUBDEVICES, ACTION_UPDATE, parameters);
+        return JsonResponseHandler.getObject(result, JsonSubdevice.class);
+    }
+
+    @Override
+    public JsonSubdevice unregisterSubdevice(int id) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DEVICE_ID_KEY, id);
+        JsonObject result = execute(CONTROLLER_SUBDEVICES, ACTION_DELETE, parameters);
+        return JsonResponseHandler.getObject(result, JsonSubdevice.class);
+    }
+
+    private JsonObject execute(String controller, String action) {
+        return execute(controller, action, new HashMap<>());
+    }
+
+    private JsonObject execute(String controller, String action, Map<String, Object> content) {
+
+        Properties httpHeaders = new Properties();
+
+        // Set basic authentication header
+        String authData = configuration.getUserName() + ":" + configuration.getPassword();
+        String basicAuthentication = "Basic " + Base64.getEncoder().encodeToString(authData.getBytes());
+        basicAuthentication = basicAuthentication.trim();
+        httpHeaders.setProperty("Authorization", basicAuthentication);
+
+        // Convert request the content to JSON
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(content);
+        InputStream requestContent = IOUtils.toInputStream(jsonContent);
+        ContentResponse contentResponse = null;
+        String responseBody = null;
+
+        try {
+            // all requests are sent using POST method, as it is recommended from the API documentation
+            contentResponse = restClient.sendRequest(controller + "/" + action, RestClient.DEFAULT_HTTP_METHOD,
+                    httpHeaders, requestContent, RestClient.CONTENT_TYPE);
+
+            byte[] rawResponse = contentResponse.getContent();
+
+            String encoding = contentResponse.getEncoding() != null
+                    ? contentResponse.getEncoding().replaceAll("\"", "").trim() : DEFAULT_RESPONSE_ENCODING;
+
+            if (!Charset.availableCharsets().keySet().contains(encoding)) {
+                logger.warn("Unsupported content encoding : {}. Fall back to the default encoding {}", encoding,
```

Comment:
Is this a bug that the user should report? Then better also include the stacktrace. If not: Of what use is that info to the user?